app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :read, User
+    can :manage, User
+    can :manage, RecipeFood
+    return unless user.present?
+
+    can :read, Recipe, public: true
+    can :manage, Recipe, user_id: user.id
+    can :read, Food
+    can :manage, Food, user_id: user.id
+  end
```

Comment:
- [ ] @team Great work adding the ability model to restrict access however you didn't add the functionality to restrict only the owner of the to have the ability to add new food to the recipe ,currently all users can access the add ingredients page also can delete ingredients